src/language/predicates.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2018-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict
+ */
+
+import type { ASTNode } from './ast';
+import { Kind } from './kinds';
+
+export function isDefinitionNode(node: ASTNode): boolean %checks {
```

Comment:
I am a big fan of these! We hit them a lot, and it's nice to be able to filter through definitions to just get `Executable` or `TypeSystem` definitions!